### PR TITLE
Feature/rbac on collection ids

### PR DESCRIPTION
**Pull request by TangoBeee**

### CODE REVIEW

File: apps/dashboard/src/main/java/com/akto/action/observe/InventoryAction.java [hunk 1]
```diff
@@ -107,6 +108,10 @@ public List<BasicDBObject> fetchRecentEndpoints(int startTimestamp, int endTimes
             List<Bson> pipeline = new ArrayList<>();
 
             pipeline.add(Aggregates.match(Filters.in("apiCollectionId", nonHostApiCollectionIds)));
+            List<Integer> collectionIds = UsersCollectionsList.getCollectionsIdForUser(Context.userId.get(), Context.accountId.get());
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/action/observe/InventoryAction.java:111]: take common items here also

File: apps/dashboard/src/main/java/com/akto/utils/RiskScoreOfCollections.java [hunk 2]
```diff
@@ -188,6 +189,10 @@ private static void writeUpdatesForSensitiveInfoInApiInfo(List<String> updatedDa
         );
         List<Bson> pipeline = new ArrayList<>();
         pipeline.add(Aggregates.match(sensitiveSubTypeFilter));
+        List<Integer> collectionIds = UsersCollectionsList.getCollectionsIdForUser(Context.userId.get(), Context.accountId.get());
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/utils/RiskScoreOfCollections.java:192]: remove from here

File: libs/dao/src/main/java/com/akto/dao/ApiCollectionsDao.java [hunk 3]
```diff
@@ -139,6 +140,10 @@ public Map<Integer, Integer> buildEndpointsCountToApiCollectionMap() {
 
         BasicDBObject groupedId = new BasicDBObject(SingleTypeInfo._COLLECTION_IDS, "$" + SingleTypeInfo._COLLECTION_IDS);
         pipeline.add(Aggregates.unwind("$" + SingleTypeInfo._COLLECTION_IDS));
+        List<Integer> collectionIds = UsersCollectionsList.getCollectionsIdForUser(Context.userId.get(), Context.accountId.get());
+        if(collectionIds != null && !collectionIds.isEmpty()) {
```
Comment (Ark2307) [inline libs/dao/src/main/java/com/akto/dao/ApiCollectionsDao.java:144]: not needed here as well

File: libs/dao/src/main/java/com/akto/dao/FilterSampleDataDao.java [hunk 4]
```diff
@@ -55,4 +57,9 @@ public String getCollName() {
     public Class<FilterSampleData> getClassT() {
         return FilterSampleData.class;
     }
+
+    @Override
+    public String getFilterKeyString() {
+        return TestingEndpoints.getFilterPrefix(ApiCollectionUsers.CollectionType.Id_ApiInfoKey_ApiCollectionId) + ApiInfo.ApiInfoKey.API_COLLECTION_ID;
```
Comment (Ark2307) [inline libs/dao/src/main/java/com/akto/dao/FilterSampleDataDao.java:63]: confirm this once

**Review by Ark2307 — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java [hunk 5]
```diff
@@ -596,7 +615,13 @@ public String updateEnvType(){
             FindOneAndUpdateOptions updateOptions = new FindOneAndUpdateOptions();
             updateOptions.upsert(false);
 
-            UpdateResult result = ApiCollectionsDao.instance.getMCollection().updateMany(filter,
+            Bson collectionFilter = Filters.empty();
+            List<Integer> collectionIds = UsersCollectionsList.getCollectionsIdForUser(Context.userId.get(), Context.accountId.get());
+            if(collectionIds != null && !collectionIds.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (TangoBeee) [reply apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java:620]: > what you have access to

I didn't get this.

File: apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java [hunk 6]
```diff
@@ -616,7 +641,7 @@ public String updateUserCollections() {
 
         for(Map.Entry<String, List<Integer>> entry : userCollectionMap.entrySet()) {
```
[the remark this answers is not in the document]
Comment (TangoBeee) [reply apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java:642]: How do I convert a list to a set in a loop?

**Pull request merged**